client.go

```diff
@@ -87,6 +102,16 @@ type Client struct {
 	cacheExpiration time.Duration
 	cachedEntries   map[string]clientCacheEntry
 	cachedEntryLock *sync.RWMutex
+	logger          Logger
+	requestLog      func(*RequestLog) error
+	onBeforeRequest []func(*http.Request) error
+	onAfterResponse []func(*http.Response) error
+
+	retryConditionals []RetryConditional
+	retryMaxWaitTime  time.Duration
+	retryMinWaitTime  time.Duration
```

Comment:
Where are `retryMaxWaitTime` and `retryMinWaitTime` applied to the http client? I can't see they are used anywhere..

Reply:
Good catch! These can actually be removed. I initially included them in the Client struct since they were used in the resty client and forgot about them. The http client has it's own timeout property.

Reply:
Actually, removing them would cause a breaking change that I would like to avoid. I'll keep thinking about this.

Reply:
Alright this should be fixed now!

Reply:
Thanks @ezilber-akamai!

Reply:
Can you also give them some default values or making them as a pointer type if not already? This can prevent [Go zero values](https://go.dev/tour/basics/12) accidentally being used as the actual configuration.

Reply:
Good call. I added default values in NewClient 👍